cpu/nrf5x_common/include/periph_cpu.h

```diff
@@ -54,15 +54,18 @@ extern "C" {
 #define GPIO_PIN(x,y)       ((x & 0) | y)
 
 /**
- * @brief   Override GPIO pull register select values
+ * @brief   Override GPIO modes
  * @{
  */
-#define HAVE_GPIO_PP_T
+#define HAVE_GPIO_MODE_T
 typedef enum {
-    GPIO_NOPULL = 0,        /**< do not use internal pull resistors */
-    GPIO_PULLUP = 3,        /**< enable internal pull-up resistor */
-    GPIO_PULLDOWN = 1       /**< enable internal pull-down resistor */
-} gpio_pp_t;
+    GPIO_IN    = (0 << 0 | 0 << 1 | 0 << 2),    /**< IN */
```

Comment:
Actually I like the shift notation as chosen quite a bit, though I forgot to write a comment above describing the bits as I did e.g. for the stm32x CPUs